src/Sulu/Bundle/RouteBundle/DependencyInjection/RouteGeneratorCompilerPass.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\RouteBundle\DependencyInjection;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Collects route-generator by configured mappings.
+ */
+class RouteGeneratorCompilerPass implements CompilerPassInterface
+{
+    const SERVICE_ID = 'sulu_route.manager.route_manager';
+    const PARAMETER_NAME = 'sulu_route.mappings';
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasDefinition(self::SERVICE_ID) || !$container->hasParameter(self::PARAMETER_NAME)) {
+            return;
+        }
+
+        $config = $container->getParameter(self::PARAMETER_NAME);
+
+        $services = [];
+        foreach ($config as $item) {
+            $services[$item['service_id']] = new Reference($item['service_id']);
```

Comment:
Shouldn't we implement that with tags as we do with most of our other components, and use an alias defined by this tag in the configuration instead of the service id?